cpp/python/CMakeLists.txt

```diff
@@ -3,13 +3,78 @@
 # SPDX-License-Identifier: BSD 3-Clause License
 # ======================================================================================================
 
+cmake_minimum_required(VERSION 3.26.4 FATAL_ERROR)
+
+include(../../fetch_rapids.cmake)
+include(rapids-cmake)
+include(rapids-cpm)
+include(rapids-export)
+include(rapids-find)
+
+file(READ "${CMAKE_CURRENT_LIST_DIR}/../../VERSION" _version_contents)
+if(_version_contents MATCHES "^([0-9]+)\\.([0-9]+)\\.([0-9]+).*$")
+  set(libucxx_version "${CMAKE_MATCH_1}.${CMAKE_MATCH_2}.${CMAKE_MATCH_3}")
+else()
+  string(REPLACE "\n" "\n  " _version_contents_formatted "${_version_contents}")
+  message(FATAL_ERROR "Could not determine ucxx version. Contents of VERSION file:\n  ${_version_contents_formatted}")
+endif()
+
+project(
+  UCXX_PYTHON
+  VERSION ${libucxx_version}
+  LANGUAGES C CXX
+)
+
+option(BUILD_SHARED_LIBS "Build UCXX Python shared libraries" ON)
+option(FIND_UCXX_CPP "Search for existing UCXX C++ installations before defaulting to local files"
+       OFF
+)
+
+# add third party dependencies using CPM
+rapids_cpm_init()
+# find rmm
+include(../cmake/thirdparty/get_rmm.cmake)
+
+if(FIND_UCXX_CPP)
+  rapids_find_package(
+    ucxx REQUIRED
+    BUILD_EXPORT_SET ucxx-python-exports
+    INSTALL_EXPORT_SET ucxx-python-exports
+  )
+else()
+  set(UCXX_ENABLE_RMM ON)
+  add_subdirectory(.. ucxx-cpp)
+endif()
+
 rapids_find_package(
   Python3 REQUIRED COMPONENTS Development
   BUILD_EXPORT_SET ucxx-python-exports
   INSTALL_EXPORT_SET ucxx-python-exports
 )
 set(UCXX_PYTHON_LIB Python3::Python)
 
+# ##################################################################################################
+# * compiler options ------------------------------------------------------------------------------
+# Due to https://github.com/openucx/ucx/issues/9614, we cannot export the ucx
+# dependency because users would then have no control over whether ucx is found
+# multiple times, causing potential configure errors. Therefore, we use a raw
+# find_package call instead of rapids_find_package and skip exporting the ucx
+# dependency. Consumers of ucxx must find ucx themselves. Once we move the
+# minimum version to UCX 1.16 we can remove the above find_package in favor of
+# the commented out lines below. For the same reason, we must also gate this
+# find_package call behind a check for the target already existing so that
+# consumers can use tools like CPM.cmake to either find or build ucxx from
+# source if it cannot be found (i.e. both cases must allow prior finding of
+# ucx).
+if(NOT TARGET ucx::ucp)
+  find_package(ucx REQUIRED)
+endif()
+#rapids_find_package(
+#  ucx REQUIRED
+#  BUILD_EXPORT_SET ucxx-exports
+#  INSTALL_EXPORT_SET ucxx-exports
+#)
```

Comment:
```suggestion
```

Reply:
Should this be removed?

Reply:
No, because `ucxx::python` has a `PUBLIC` dependency on `ucx::ucp` and thus needs to export it (once the minimum is 1.16).